enarx-keep/build.rs

```diff
@@ -32,34 +34,94 @@ fn main() {
     println!("cargo:rerun-if-env-changed=OUT_DIR");
     println!("cargo:rerun-if-env-changed=PROFILE");
 
-    let out_dir = std::env::var("OUT_DIR").unwrap();
+    let out_dir = PathBuf::from(std::env::var("OUT_DIR").unwrap());
+    let out_dir_bin = out_dir.join("bin");
+
+    match std::fs::create_dir(&out_dir_bin) {
+        Err(e) if e.kind() == std::io::ErrorKind::AlreadyExists => {}
+        Err(e) => {
+            eprintln!("Can't create {:#?} : {:#?}", out_dir_bin, e);
+            std::process::exit(1);
+        }
+        Ok(_) => {}
+    }
 
     let profile: &[&str] = match std::env::var("PROFILE").unwrap().as_str() {
-        "debug" => &["--debug"],
+        "release" => &["--release"],
         _ => &[],
     };
 
+    let target_name = "x86_64-unknown-linux-musl";
+
+    let filtered_env: HashMap<String, String> = std::env::vars()
+        .filter(|&(ref k, _)| {
+            k == "TERM" || k == "TZ" || k == "LANG" || k == "PATH" || k == "RUSTUP_HOME"
+        })
+        .collect();
+
     for entry in std::fs::read_dir("shims").unwrap() {
-        let path: String = entry
+        let shim_path = entry.unwrap().path();
+        let shim_name = shim_path.clone();
+        let shim_name = shim_name
+            .file_name()
             .unwrap()
-            .path()
-            .into_os_string()
+            .to_os_string()
             .into_string()
             .unwrap();
 
+        let shim_out_dir = out_dir.join(&shim_path);
+
+        let target_dir = shim_out_dir.clone().into_os_string().into_string().unwrap();
+
+        let path: String = shim_path.into_os_string().into_string().unwrap();
+
         println!("cargo:rerun-if-changed={}/Cargo.toml", path);
         println!("cargo:rerun-if-changed={}/Cargo.lock", path);
         println!("cargo:rerun-if-changed={}/link.json", path);
         rerun_src(&path).unwrap();
 
-        Command::new("cargo")
-            .arg("install")
+        let stdout: Stdio = OpenOptions::new()
+            .write(true)
+            .open("/dev/tty")
+            .map(Stdio::from)
+            .unwrap_or_else(|_| Stdio::inherit());
+
+        let stderr: Stdio = OpenOptions::new()
+            .write(true)
+            .open("/dev/tty")
+            .map(Stdio::from)
+            .unwrap_or_else(|_| Stdio::inherit());
+
+        let status = Command::new("cargo")
```

Comment:
We probably need to use `std::env::var("CARGO")` here.

Reply:
can be a follow-up PR